scm-webapp/src/main/java/sonia/scm/api/v2/resources/RepositoryImportResource.java

```diff
@@ -0,0 +1,548 @@
+//                                   @QueryParam("compressed")
+//                                   @DefaultValue("false") boolean compressed) {
+//    Repository repository = doImportFromBundle(type, namespace, name, inputStream, compressed);
+//
+//    return buildResponse(uriInfo, repository);
+//  }
+//
+//  /**
+//   * This method works exactly like
+//   * {@link #importFromBundle(UriInfo, String, String, String, InputStream, boolean)}, but this
+//   * method returns an html content-type. The method exists only for a
+//   * workaround of the javascript ui extjs. <strong>Note:</strong> This method requires admin
+//   * privileges.
+//   *
+//   * @param type        repository type
+//   * @param name        name of the repository
+//   * @param inputStream input bundle
+//   * @param compressed  true if the bundle is gzip compressed
+//   * @return empty response with location header which points to the imported
+//   * repository
+//   * @since 1.43
+//   */
+//  @POST
+//  @Path("{type}/bundle.html")
+//  @Consumes(MediaType.MULTIPART_FORM_DATA)
+//  @Produces(MediaType.TEXT_HTML)
+//  public Response importFromBundleUI(@PathParam("type") String type,
+//                                     @FormParam("namespace") String namespace,
+//                                     @FormParam("name") String name,
+//                                     @FormParam("bundle") InputStream inputStream,
+//                                     @QueryParam("compressed")
+//                                     @DefaultValue("false") boolean compressed) {
+//    Response response;
+//
+//    try {
+//      doImportFromBundle(type, namespace, name, inputStream, compressed);
+//      response = Response.ok(new RestActionUploadResult(true)).build();
+//    } catch (WebApplicationException ex) {
+//      logger.warn("error durring bundle import", ex);
+//      response = Response.fromResponse(ex.getResponse()).entity(
+//        new RestActionUploadResult(false)).build();
+//    }
+//
+//    return response;
+//  }
+
+  /**
+   * Imports a external repository which is accessible via url. The method can
+   * only be used, if the repository type supports the {@link Command#PULL}. The
+   * method will return a location header with the url to the imported
+   * repository.
+   *
+   * @param uriInfo uri info
+   * @param type    repository type
+   * @param request request object
+   * @return empty response with location header which points to the imported
+   * repository
+   * @since 2.11.0
+   */
+  @POST
+  @Path("{type}/url")
+  @Consumes(VndMediaType.REPOSITORY)
+  @Operation(summary = "Import repository from url", description = "Imports the repository for the given url.", tags = "Repository")
+  @ApiResponse(
+    responseCode = "201",
+    description = "Repository import was successful",
+    content = @Content(
+      mediaType = VndMediaType.REPOSITORY,
+      schema = @Schema(implementation = RepositoryDto.class)
```

Comment:
Is RepositoryDto the correct schema type? Shouldn't this be `RepositoryImportDto`, but without `_links`? Do we need an interface for the schema?